payment_methods/SquareOnsite/forms/SettingsForm.php

```diff
@@ -0,0 +1,226 @@
+                    'html_label_text' => sprintf(
+                        // translators: %1$s: Help tab link as icon.
+                        esc_html__('Application ID %1$s', 'event_espresso'),
+                        $paymentMethod->get_help_tab_link()
+                    ),
+                    'html_id'         => $pmInstance->slug() . '-app-id',
+                    'required'        => true,
+                ]),
+                Domain::META_KEY_ACCESS_TOKEN => new EE_Password_Input([
+                    'html_label_text' => sprintf(
+                        // translators: %1$s: Help tab link as icon.
+                        esc_html__('Access Token %1$s', 'event_espresso'),
+                        $paymentMethod->get_help_tab_link()
+                    ),
+                    'html_id'         => $pmInstance->slug() . '-access-token',
+                    'required'        => true,
+                ]),
+                Domain::META_KEY_USE_DIGITAL_WALLET => new EE_Yes_No_Input(
+                    [
+                        'html_label_text' => sprintf(
+                            // translators: %1$s: Help tab link as icon.
+                            esc_html__('Enable Digital Wallet ? %s', 'event_espresso'),
+                            $paymentMethod->get_help_tab_link()
+                        ),
+                        'html_help_text'  => esc_html__(
+                            // @codingStandardsIgnoreStart
+                            'Would you like to enable Google Pay and Apple Pay as payment options on the checkout page ?',
+                            // @codingStandardsIgnoreEnd
+                            'event_espresso'
+                        ),
+                        'html_id'         => $pmInstance->slug() . '-use-dwallet',
+                        'default'         => false,
+                        'required'        => true,
+                    ]
+                ),
+                Domain::META_KEY_LOCATION_ID => new EE_Text_Input([
+                    'html_label_text' => sprintf(
+                        // translators: %1$s: Help tab link as icon.
+                        esc_html__('Location ID %1$s', 'event_espresso'),
+                        $paymentMethod->get_help_tab_link()
+                    ),
+                    'html_id'         => $pmInstance->slug() . '-location-id',
+                    'required'        => true,
+                ]),
+            ]
+        ];
+        // Build the PM form.
+        parent::__construct($pmFormParams);
+
+        // Now add the OAuth section.
+        $this->addSquareConnectButton($paymentMethod, $pmInstance);
+    }
+
+
+    /**
+     *  Add the connect button to the PM settings page.
+     *
+     * @param EE_PMT_SquareOnsite    $paymentMethod
+     * @param EE_Payment_Method      $pmInstance
+     * @return void
+     * @throws EE_Error|ReflectionException
+     */
+    public function addSquareConnectButton($paymentMethod, $pmInstance)
```

Comment:
should type hint for safety

```suggestion
    public function addSquareConnectButton(EE_PMT_SquareOnsite $paymentMethod, EE_Payment_Method $pmInstance)
```